src/Http/Http.Abstractions/src/RequestDelegateWithMetadata.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Http
+{
+    /// <summary>
+    /// A Class that represents RequestDelegate with associated metadata.
+    /// </summary>
+
+    public sealed class RequestDelegateResult
+    {
+        /// <summary>
+        /// A function that can process an HTTP request.
+        /// </summary>
+        /// <returns>A task that represents the completion of request processing.</returns>
+
+        public RequestDelegate? RequestDelegate { get; init; }
```

Comment:
Can we add a constructor so that this isn't nullable?

Reply:
Added.